tests/data/cases/pattern_matching_with_if_stmt.py

```diff
@@ -0,0 +1,56 @@
+# flags: --preview --minimum-version=3.10
+match "test":
```

Comment:
Can you add some test cases where there are comments inside existing parentheses? That often causes us grief. Something like this:

```
match match:
    case case if (True # some comment
    ):
        pass
```

Reply:
Note the `with line-length=1` part. That happens in production too: https://black.vercel.app/?version=main&state=_Td6WFoAAATm1rRGAgAhARYAAAB0L-Wj4ACgAGddAD2IimZxl1N_WmBasjhgIp6QLGDHMEqd7_9GPuZidp8Xgwq_Pjz0403YCzqV5LN_bzUhCM4zzUe4MOiUzBZKKVWtbCBHDK1ePRRxhpRUs8epyuKOCYVT26y790ipGGDrjrDj2VLCkQAAACeMpDur05NRAAGDAaEBAABMjwSSscRn-wIAAAAABFla

I make it run all tests with `--line-length=1` because it's a good way to find additional crashes. Such crashes can often also appear with a more reasonable line length if the code is deeply nested.

Reply:
Looks like there are still no tests with comments inside the if parentheses. I think such tests are important because comments in inopportune places often lead to crashes in Black.

Reply:
Thanks for the insights! Yes, I haven't committed any new tests yet. The latest commit was just to incorporate your suggestions on descriptions.

I did test a few cases like the ones below, and everything seems to work fine except for the `case case if True:` test, this one would crush even without the modifications on this PR (the crushed test I mentioned above was tested on current main branch). I need some more time to figure out why the `case case if (...)` caused a crash in Black's tests. I can open a new issue for this specific crash and write a PR later if you agree (since it's a crash not directly related to this issue) so that the fix for this issue can be merged. Alternatively I can investigate a bit more to incorporate this `case case if (...)` test in this PR.

```
    case "test" if (True  # some comment
        ):
        pass  # some comment
    case cases if (True  # some comment
                   ):
        pass  # some comment
    case match if (True  # some comment
                   ):
        pass  # some comment
    case case if True:
        pass
```

Reply:
Sure, if it's not new to this PR, it's fine to open a new issue. Thanks for investigating!

Reply:
Gotcha, I'll open a new issue and commit the new tests soon. Thanks!